### PR TITLE
spotube: 3.7.1 -> 3.8.0

### PR DESCRIPTION
**Pull request by nebunebu**
## Description of changes

This PR bumps the spotube package version.
https://github.com/KRTirtho/spotube/releases/tag/v3.8.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by TomaSajt** [comment]
Thanks for opening this PR.
I'm currently not able to test this since I'm away from home this week, but I'll try to review it when I get back.

**Comment by bbigras** [comment]
Result of `nixpkgs-review pr 339705` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>spotube</li>
  </ul>
</details>

**Review by bbigras — APPROVED** [review]

**Comment by bbigras** [comment]
Builds and runs fine, thanks!

**Review by TomaSajt — CHANGES_REQUESTED** [review]
I tried reducing the changes made to the app and I found that I only needed to add `webkitgtk_4_1` `libsoup_3` and `gtk3` to the `buildInputs`. Also, `libsecret` could be removed since it's probably propagated by one of the new dependencies.

None of the changes made to the call of `makeWrapper` were necessary in my experience.

Also, the hash for the darwin source was not valid.
I had to regenerate the hash by running `nix-build -A pkgsCross.x86_64-darwin.spotube.src`. Did upstream change the file? I'm not sure.

Please do explain why you added all those other dependencies, as I might be missing something here.

This is the minimal working diff I came up with:
```patch
diff --git a/pkgs/by-name/sp/spotube/package.nix b/pkgs/by-name/sp/spotube/package.nix
index f2f9cbc23d27..bfeade5ef044 100644
--- a/pkgs/by-name/sp/spotube/package.nix
+++ b/pkgs/by-name/sp/spotube/package.nix
@@ -10,16 +10,18 @@
   undmg,
   wrapGAppsHook3,

+  gtk3,
   libappindicator,
   libnotify,
-  libsecret,
+  libsoup_3,
   mpv-unwrapped,
+  webkitgtk_4_1,
   xdg-user-dirs,
 }:

 let
   pname = "spotube";
-  version = "3.7.1";
+  version = "3.8.0";

   meta = {
     description = "Open source, cross-platform Spotify client compatible across multiple platforms";
@@ -53,7 +55,7 @@ let

     src = fetchArtifact {
       filename = "Spotube-macos-universal.dmg";
-      hash = "sha256-EYgjVXO/ztIsVYzEHe14YgXbQTclQIht9Qqr8ewHU8w=";
+      hash = "sha256-qQDbGRnia8JAclm2AgT2FCxhYS6WoNuDWIMbG76pDB0=";
     };

     sourceRoot = ".";
@@ -77,7 +79,7 @@ let

     src = fetchArtifact {
       filename = "Spotube-linux-x86_64.deb";
-      hash = "sha256-JKp2RMYNfdBzywqlBpTaHL1iD+E71EL8xY+nzkdA3us=";
+      hash = "sha256-xgwHRaFeQ182kRhUzCEvMx56WyPnHu8aCDyQ5wzVKRw=";
     };

     nativeBuildInputs = [
@@ -88,10 +90,12 @@ let
     ];

     buildInputs = [
+      gtk3
       libappindicator
       libnotify
-      libsecret
+      libsoup_3
       mpv-unwrapped
+      webkitgtk_4_1
     ];

     dontWrapGApps = true;
```

**Comment by nebunebu** [comment]
You're not missing anything. I appreciate the feedback. I grabbed the sha for darwin with `nurl https://github.com/KRTirtho/spotube/releases/download/v3.8.0/Spotube-macos-universal.dmg`, but did not test.

File: pkgs/by-name/sp/spotube/package.nix [hunk 1]
```diff
@@ -111,8 +116,19 @@ let
...
+              gtk3
+              libGL
+              libsoup_3
+              mesa
+              mpv-unwrapped
+              webkitgtk_4_1
+            ]
+          } \
+          --prefix PATH : ${lib.makeBinPath [ xdg-user-dirs ]} \
+          --set LIBGL_ALWAYS_SOFTWARE 1
```
Comment (TomaSajt) [inline pkgs/by-name/sp/spotube/package.nix:131]: For me, only `mpv-unwrapped` is needed inside `makeLibraryPath`'s list.

Also, `LIBGL_ALWAYS_SOFTWARE` doesn't seem to be needed for me...
Is this perhaps some X11/wayland difference?

File: pkgs/by-name/sp/spotube/package.nix [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
+  libappindicator,
+  libnotify,
+  libsoup_3,
   makeBinaryWrapper,
   makeWrapper,
+  mesa,
+  mpv-unwrapped,
+  stdenv,
   undmg,
+  webkitgtk_4_1,
```
Comment (TomaSajt) [inline pkgs/by-name/sp/spotube/package.nix:18]: Please restore the changes made to the ordering, as it's not necessary for this PR.

I generally only sort the `nativeBuildInputs` and `buildInputs` among themselves, not the whole list.

Comment (nebunebu) [reply pkgs/by-name/sp/spotube/package.nix]: The ordering should be restored, everything except `mpv-unwrapped` is removed from `mkLibraryPath`, and `LIBGL_ALWAYS_SOFTWARE` removed. I am on wayland and it still built and ran with this modifications.

File: pkgs/by-name/sp/spotube/package.nix [hunk 3]
```diff
@@ -112,7 +116,7 @@ let
       makeWrapper $out/share/spotube/spotube $out/bin/spotube \
           "''${gappsWrapperArgs[@]}" \
           --prefix LD_LIBRARY_PATH : $out/share/spotube/lib:${lib.makeLibraryPath [ mpv-unwrapped ]} \
-          --prefix PATH : ${lib.makeBinPath [ xdg-user-dirs ]}
+          --prefix PATH : ${lib.makeBinPath [ xdg-user-dirs ]} \
```
Comment (TomaSajt) [inline pkgs/by-name/sp/spotube/package.nix:119]:
```suggestion
          --prefix PATH : ${lib.makeBinPath [ xdg-user-dirs ]}
```

**Review by TomaSajt — APPROVED** [review]
LGTM

**Comment by nebunebu** [comment]
Thank you for your patience and guidance

**Comment by Airradda** [comment]
I believe a dependency or some configuration may have been missed as the web view window used for the Spotify login gives the error `TLS support is not available`. Considering how long this release has been out without a bug report upstream, unless I missed it, I believe it is something here.

**Comment by nebunebu** [comment]
@Airradda, thanks. I was logged in from use of a previous version. When I logged out and attempted to log back in, I also get `TLS support is not available` in the popup window in place of the login prompts.

**Comment by nebunebu** [comment]
I'm not sure whether this should be opened in a new issue or continued here.

If you built and run with `LD_DEBUG=libs ./result/bin/spotube`, when clicking the login button for spotify, the application outputs,

```txt
** (spotube:2070646): CRITICAL **: 15:05:18.929: void fl_method_channel_invoke_method(FlMethodChannel *, const gchar *, FlValue *, GCancellable *, GAsyncReadyCallback, gpointer): assertion 'FL_IS_METHOD_CHANNEL(self)' failed
   2071387:     find library=libGLX.so.1 [0]; searching
   2071387:      search path=/nix/store/fglh7b2imdi4hpk0l7x7vjmlb7rsq5z7-spotube-3.8.0/share/spotube/lib:/nix/store/raqpmam9sxp29hvlb7wpy6gmvmws8ppl-mpv-0.37.0/lib:/nix/store/4qh6habb6p26f0nwrhf13rqj0h6kp7x2-pipewire-1.0.7-jack/lib             (LD_LIBRARY_PATH)
   2071387:       trying file=/nix/store/fglh7b2imdi4hpk0l7x7vjmlb7rsq5z7-spotube-3.8.0/share/spotube/lib/libGLX.so.1
   2071387:       trying file=/nix/store/raqpmam9sxp29hvlb7wpy6gmvmws8ppl-mpv-0.37.0/lib/libGLX.so.1
   2071387:       trying file=/nix/store/4qh6habb6p26f0nwrhf13rqj0h6kp7x2-pipewire-1.0.7-jack/lib/libGLX.so.1
   2071387:      search path=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/lib   (system search path)
   2071387:       trying file=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/lib/libGLX.so.1
   2071387:      search cache=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/etc/ld.so.cache
   2071387:      search path=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/lib:/nix/store/k8aiaw3mslh4120lah1ssg3r6xa46cz1-xgcc-13.2.0-libgcc/lib         (system search path)
   2071387:       trying file=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/lib/libGLX.so.1
   2071387:       trying file=/nix/store/k8aiaw3mslh4120lah1ssg3r6xa46cz1-xgcc-13.2.0-libgcc/lib/libGLX.so.1
   2071387:
   2071387:     find library=libGL.so.1 [0]; searching
   2071387:      search path=/nix/store/fglh7b2imdi4hpk0l7x7vjmlb7rsq5z7-spotube-3.8.0/share/spotube/lib:/nix/store/raqpmam9sxp29hvlb7wpy6gmvmws8ppl-mpv-0.37.0/lib:/nix/store/4qh6habb6p26f0nwrhf13rqj0h6kp7x2-pipewire-1.0.7-jack/lib             (LD_LIBRARY_PATH)
   2071387:       trying file=/nix/store/fglh7b2imdi4hpk0l7x7vjmlb7rsq5z7-spotube-3.8.0/share/spotube/lib/libGL.so.1
   2071387:       trying file=/nix/store/raqpmam9sxp29hvlb7wpy6gmvmws8ppl-mpv-0.37.0/lib/libGL.so.1
   2071387:       trying file=/nix/store/4qh6habb6p26f0nwrhf13rqj0h6kp7x2-pipewire-1.0.7-jack/lib/libGL.so.1
   2071387:      search path=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/lib   (system search path)
   2071387:       trying file=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/lib/libGL.so.1
   2071387:      search cache=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/etc/ld.so.cache
   2071387:      search path=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/lib:/nix/store/k8aiaw3mslh4120lah1ssg3r6xa46cz1-xgcc-13.2.0-libgcc/lib         (system search path)
   2071387:       trying file=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/lib/libGL.so.1
   2071387:       trying file=/nix/store/k8aiaw3mslh4120lah1ssg3r6xa46cz1-xgcc-13.2.0-libgcc/lib/libGL.so.1
   2071387:
   2071387:     find library=libGLX.so.1 [0]; searching
   2071387:      search path=/nix/store/fglh7b2imdi4hpk0l7x7vjmlb7rsq5z7-spotube-3.8.0/share/spotube/lib:/nix/store/raqpmam9sxp29hvlb7wpy6gmvmws8ppl-mpv-0.37.0/lib:/nix/store/4qh6habb6p26f0nwrhf13rqj0h6kp7x2-pipewire-1.0.7-jack/lib             (LD_LIBRARY_PATH)
   2071387:       trying file=/nix/store/fglh7b2imdi4hpk0l7x7vjmlb7rsq5z7-spotube-3.8.0/share/spotube/lib/libGLX.so.1
   2071387:       trying file=/nix/store/raqpmam9sxp29hvlb7wpy6gmvmws8ppl-mpv-0.37.0/lib/libGLX.so.1
   2071387:       trying file=/nix/store/4qh6habb6p26f0nwrhf13rqj0h6kp7x2-pipewire-1.0.7-jack/lib/libGLX.so.1
   2071387:      search path=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/lib   (system search path)
   2071387:       trying file=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/lib/libGLX.so.1
   2071387:      search cache=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/etc/ld.so.cache
   2071387:      search path=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/lib:/nix/store/k8aiaw3mslh4120lah1ssg3r6xa46cz1-xgcc-13.2.0-libgcc/lib         (system search path)
   2071387:       trying file=/nix/store/r8qsxm85rlxzdac7988psm7gimg4dl3q-glibc-2.39-52/lib/libGLX.so.1
   2071387:       trying file=/nix/store/k8aiaw3mslh4120lah1ssg3r6xa46cz1-xgcc-13.2.0-libgcc/lib/libGLX.so.1
```
Merely adding `libGL` and/or `libGLU` and/or `openssl` to buildInputs does not resolve the issue.

When adding them, I am not finding a `libGLX.so` or `libGL.so` under `./result/share` either.

**Comment by TomaSajt** [comment]
Yeah, I see that it's not working for me either...

I found that adding `glib-networking` to `buildInputs` makes it work for me
(it's not used by `autoPatchelfHook`, but `wrapGAppsHook3`)

**Comment by nebunebu** [comment]
@TomaSajt adding `glib-networking` also worked for me

**Pull request merged**